pkg/agent/openflow/client.go

```diff
@@ -647,6 +666,41 @@ func (c *client) InstallExternalFlows() error {
 	return nil
 }
 
+func (c *client) InstallSNATMarkFlows(snatIP net.IP, mark uint32) error {
+	c.replayMutex.RLock()
+	defer c.replayMutex.RUnlock()
+	flows := c.snatMarkFlows(snatIP, mark)
+	cacheKey := fmt.Sprintf("s%x", mark)
+	return c.addFlows(c.snatFlowCache, cacheKey, flows)
+}
+
+func (c *client) UninstallSNATMarkFlows(mark uint32) error {
+	c.replayMutex.RLock()
+	defer c.replayMutex.RUnlock()
+	cacheKey := fmt.Sprintf("s%x", mark)
+	return c.deleteFlows(c.snatFlowCache, cacheKey)
+}
+
+func (c *client) InstallPodSNATFlows(ofPort uint32, snatMark uint32, isIPv6 bool) error {
+	c.replayMutex.RLock()
+	defer c.replayMutex.RUnlock()
+	flows := make([]binding.Flow, 0, 2)
```

Comment:
why make it 2? I see only one flow

Reply:
Fixed it.
Originally I thought we might need to add a flow for both v4 and v6 in the dual stack case.